src/main/java/Parser.java

```diff
@@ -0,0 +1,234 @@
+    private ArrayList<Task> items;
+
+    /**
+     * Constructor method for Parser.
+     * @param user object handling all messages
+     */
+    public Parser(Ui user) {
+        this.user = user;
+        this.items = new ArrayList<>();
+    }
+
+    //return Ui obj
+    public Ui getUser() {
+        return this.user;
+    }
+
+    //return list of tasks
+    public ArrayList<Task> getList() {
+        return this.items;
+    }
+
+    /**
+     * Deleting of task from ArrayList.
+     * @param num index of task in list
+     * @return String delete message
+     * @throws IndexOutOfBoundsException index of task not found in list
+     *
+     */
+    public String deleteTask(int num) throws IndexOutOfBoundsException {
+        Task curr = items.get(num - 1);
+        items.remove(num - 1);
+        String reply = user.deleteMessage(curr, items.size());
+        return reply;
+    }
+
+    /**
+     * Converting String to datetime for storage.
+     * @param input String from user input containing datetime for deadline
+     * @return Date containing dd/MM/yyyy HH:mm
+     * @throws ParseException when input does not matched date format
+     */
+    public static Date convertStringToDeadline(String input) throws ParseException {
+        Date result = new SimpleDateFormat("dd/MM/yyyy HH:mm").parse(input);
+        return result;
+    }
+
+    /**
+     * Converting String to datetime for storage.
+     * @param input String from user input from date to start time for event
+     * @return Date containing dd/MM/yyy HH:mm
+     * @throws ParseException input does not matched date format
+     */
+    public static Date convertStringToEventStart(String input) throws ParseException {
+        Date result = new SimpleDateFormat("dd/MM/yyyy HH:mm").parse(input);
+        return result;
+    }
+
+    /**
+     * Converting String to datetime for storage.
+     * @param input String from user input containing end time for event
+     * @return Date containing HH:mm
+     * @throws ParseException input does not matched date format
+     */
+    public static Date convertStringToEventEnd(String input) throws ParseException {
+        Date result = new SimpleDateFormat("HH:mm").parse(input);
+        return result;
+    }
+
+    /**
+     * Generation of task - Todo, Event, Deadline and store in ArrayList.
+     * @param input String input from user input
+     * @return output message for different task
+     */
+    public String generateTask(String input) {
+        StringBuilder sb = new StringBuilder();
+        String message = "";
+        if (input.toLowerCase().contains("todo")) {
+            //adding an item
+            try {
+                Todo newTask = new Todo(input.substring(5));
+                if (!checkDuplicate(newTask)) {
+                    items.add(newTask);
+                    message = user.generateMessage(newTask, items.size());
+                    System.out.println(message);
+                } else {
```

Comment:
This might be too nested.
Avoid arrowheads.
Consider SLAPing, and make the happy path obvious.